src/osvr/Util/Log.cpp.in

```diff
@@ -0,0 +1,131 @@
+/** @file
+    @brief Implementation
+
+    @date 2016
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com>
+
+*/
+
+// Copyright 2016 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include <osvr/Util/Log.h>
+#include <osvr/Util/LogSinks.h>
+#include <osvr/Util/PlatformConfig.h>
+#include <osvr/Util/GetEnvironmentVariable.h>
+
+// Library/third-party includes
+#include <spdlog/spdlog.h>
+#include <boost/filesystem.hpp>
+
+// Standard includes
+#include <memory>           // for std::make_shared, std::shared_ptr
+#include <string>           // for std::string
+
+#cmakedefine OSVR_UTIL_LOG_SINGLETON
```

Comment:
Eeew, so can we put most of this in a normal cpp and just configure the define? A big configured file like this is a pain to maintain.